workflows/MODEL_TO_DATA.nf

```diff
@@ -53,24 +53,49 @@ include { VALIDATE } from '../modules/validate_model_to_data.nf'
 include { SCORE_MODEL_TO_DATA as SCORE } from '../modules/score_model_to_data.nf'
 include { ANNOTATE_SUBMISSION as ANNOTATE_SUBMISSION_AFTER_VALIDATE } from '../modules/annotate_submission.nf'
 include { ANNOTATE_SUBMISSION as ANNOTATE_SUBMISSION_AFTER_SCORE } from '../modules/annotate_submission.nf'
+include { ANNOTATE_SUBMISSION as ANNOTATE_SUBMISSION_AFTER_UPDATE_FOLDERS } from '../modules/annotate_submission.nf'
 include { SEND_EMAIL } from '../modules/send_email.nf'
 
 workflow MODEL_TO_DATA {
+
+    // Phase 0: Each submission is evaluated in its own separate channel
     submission_ch = CREATE_SUBMISSION_CHANNEL()
+
+    // Phase 1: Prepare the data for scoring and create output folders on Synapse
     SYNAPSE_STAGE_DATA(params.data_folder_id, "input")
     SYNAPSE_STAGE_GOLDSTANDARD(params.goldstandard_id, "goldstandard_${params.goldstandard_id}")
     CREATE_FOLDERS(submission_ch, params.project_name, params.private_folders)
     UPDATE_SUBMISSION_STATUS_BEFORE_RUN(submission_ch, "EVALUATION_IN_PROGRESS")
-    RUN_DOCKER(submission_ch, params.container_timeout, params.poll_interval, SYNAPSE_STAGE_DATA.output, params.cpus, params.memory, params.log_max_size, CREATE_FOLDERS.output, UPDATE_SUBMISSION_STATUS_BEFORE_RUN.output)
-    UPDATE_FOLDERS(submission_ch, params.project_name, RUN_DOCKER.output.map { it[1] }, RUN_DOCKER.output.map { it[2] })
-    UPDATE_SUBMISSION_STATUS_AFTER_RUN(RUN_DOCKER.output.map { it[0] }, "ACCEPTED")
-    VALIDATE(RUN_DOCKER.output, SYNAPSE_STAGE_GOLDSTANDARD.output, UPDATE_SUBMISSION_STATUS_AFTER_RUN.output, params.execute_validation)
-    UPDATE_SUBMISSION_STATUS_AFTER_VALIDATE(submission_ch, VALIDATE.output.map { it[2] })
-    ANNOTATE_SUBMISSION_AFTER_VALIDATE(VALIDATE.output)
-    SCORE(VALIDATE.output, SYNAPSE_STAGE_GOLDSTANDARD.output, UPDATE_SUBMISSION_STATUS_AFTER_VALIDATE.output, ANNOTATE_SUBMISSION_AFTER_VALIDATE.output, params.execute_scoring)
-    UPDATE_SUBMISSION_STATUS_AFTER_SCORE(submission_ch, SCORE.output.map { it[2] })
-    ANNOTATE_SUBMISSION_AFTER_SCORE(SCORE.output)
+
+    // Phase 2: Running the Docker submission (runs after Phase 1 data staging)
+    run_docker_outputs = RUN_DOCKER(submission_ch, params.container_timeout, params.poll_interval, SYNAPSE_STAGE_DATA.output, params.cpus, params.memory, params.log_max_size, CREATE_FOLDERS.output, UPDATE_SUBMISSION_STATUS_BEFORE_RUN.output)
+    //// Explicit output handling
+    run_docker_submission = run_docker_outputs.map { submission_id, predictions, logs -> submission_id }
+    run_docker_files = run_docker_outputs.map { submission_id, predictions, logs -> tuple(predictions, logs) }
```

Comment:
What is your rationale for doing these `map` steps instead of just changing the `RUN_DOCKER` outputs to match what is needed for both cases? (Same comment/question for the similar patterns below)

Reply:
The motivation was to improve readability in the workflow. When mapping the output directly within the process call (e.g. `RUN_DOCKER.map.{ it[0] }` ), it added an extra step of having to go into `run_docker.nf` to read the output getting indexed. This way helps minimize overhead when troubleshooting and understanding the pipeline.